src/network.rs

```diff
@@ -172,44 +184,62 @@ unsafe fn _network_open(
     // RANGE header definition is "bytes=x-y"
     let hvalue = format!("bytes={offset}-{end}");
     // Create a new client that can be reused for subsequent queries
-    let clt = Agent::new();
-    let req = clt.get(&url);
-    let with_headers = req.set("Range", &hvalue).set("Client", CLIENT);
-    let in_case_of_error = with_headers.clone();
-    let mut res = with_headers.call()?;
+    let clt = Agent::new_with_defaults();
+    let req = clt
+        .get(&url)
+        .header("Range", &hvalue)
+        .header("Client", CLIENT);
+
+    let mut res = req.call()?;
+
+    // Define headers for potential retries
+    let headers = [("Range", hvalue.as_str()), ("Client", CLIENT)];
+
     // hand the response off to the error-handler, continue on success
-    error_handler(&mut res, in_case_of_error)?;
+    error_handler(&mut res, &url, &headers, clt.clone())?;
+
     // Write the initial read length value into the pointer
-    let Some(Ok(contentlength)) = res.header("Content-Length").map(str::parse::<usize>) else {
-        return Err(ProjError::ContentLength);
-    };
+    let contentlength = res
+        .headers()
+        .get("Content-Length")
+        .and_then(|val| val.to_str().ok())
+        .and_then(|s| s.parse::<usize>().ok())
+        .ok_or(ProjError::ContentLength)?;
+
     let headers = res
-        .headers_names()
-        .into_iter()
+        .headers()
+        .keys()
```

Comment:
Here we're iterating over the keys and looking up the values again in the `filter_map` below.

Reply:
Good catches. Simplified with an iterator now.